test/unit/test_elasticache_redis.py

```diff
@@ -0,0 +1,222 @@
+from grafanalib.core import (
+    Alert,
+    Graph,
+    Template,
+    Templating,
+)
+
+from grafanalib.cloudwatch import CloudwatchMetricsTarget
+
+from lib.elasticache_redis import (
+    generate_elasticache_redis_connections_graph,
+    generate_elasticache_redis_cpu_usage_graph,
+    generate_elasticache_redis_latency_graph,
+    generate_elasticache_redis_memory_usage_graph,
+    generate_elasticache_redis_network_in_graph,
+    generate_elasticache_redis_network_out_graph,
+    generate_elasticache_redis_replication_graph,
+    generate_elasticache_redis_dashboard,
+)
+
+
+class TestElasticCacheRedisDashboard:
+    def test_should_generate_elasticache_redis_memory_usage_graph(self):
+
+        cache_cluster_id = "1234567890"
+        cloudwatch_data_source = "cw"
+        generated_graph = generate_elasticache_redis_memory_usage_graph(
+            cache_cluster_id=cache_cluster_id,
+            cloudwatch_data_source=cloudwatch_data_source,
+        )
+        generated_graph.should.be.a(Graph)
+        generated_graph.title.should.match(r"Memory usage")
+        generated_graph.dataSource.should.match(cloudwatch_data_source)
+        generated_graph.targets.should.have.length_of(4)
+
+        for target in generated_graph.targets:
+            target.should.be.a(CloudwatchMetricsTarget)
+            target.period.should.equal("1m")
+            target.statistics.should.equal(["Maximum"])
+            target.dimensions.should.equal({"CacheClusterId": cache_cluster_id})
+
+        generated_graph.targets[0].metricName.should.equal("BytesUsedForCache")
```

Comment:
Can you please add tests cases for alert notifications?

https://github.com/comtravo/grafana-dashboards/blob/585985fbc3e68082045b47701deecadc16519c2b/lib/elasticache_redis.py#L223-L240